src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -100,18 +100,38 @@ public boolean hasMusician(Musician musician) {
     @Override
     public void deleteMusician(Musician target) {
         addressBook.removeMusician(target);
+
+        updateFilteredMusicianList(PREDICATE_SHOW_ALL_MUSICIANS);
+        updateFilteredBandList(PREDICATE_SHOW_ALL_BANDS);
+
+        updateMusicianInAllBands(target, null);
     }
 
     @Override
     public void addMusician(Musician musician) {
         addressBook.addMusician(musician);
         updateFilteredMusicianList(PREDICATE_SHOW_ALL_MUSICIANS);
+        updateFilteredBandList(Model.PREDICATE_SHOW_ALL_BANDS);
     }
 
     @Override
     public void setMusician(Musician target, Musician editedMusician) {
         requireAllNonNull(target, editedMusician);
         addressBook.setMusician(target, editedMusician);
+
+        updateFilteredMusicianList(PREDICATE_SHOW_ALL_MUSICIANS);
+        updateFilteredBandList(PREDICATE_SHOW_ALL_BANDS);
```

Comment:
I see the rationale here. This is to jump back to the state where all the musicians and bands are listed to avoid any inconsistency and confusion for the user. Though this might not be the optimal solution, I think this is the best we can have right now.